editor/plugins/spatial_editor_plugin.cpp

```diff
@@ -5798,21 +5798,31 @@ void SpatialEditor::snap_selected_nodes_to_floor() {
 	for (List<Node *>::Element *E = selection.front(); E; E = E->next()) {
 		Spatial *sp = Object::cast_to<Spatial>(E->get());
 		if (sp) {
-			Vector3 from = Vector3();
+			Vector3 from = sp->get_global_transform().origin;
```

Comment:
As this code is now, this value will be discarded if `cs.size()` and `found_valid_shape`, or if `cs.size()` is false and `vi.size()`. This wasn't the case before.

Reply:
oops! correct, it is called snap_selected_nodes_to_floor(), not ground. I created an issue with a sample project here: https://github.com/godotengine/godot/issues/44231

I didn't see the code in the master branch at first since it changed source files. here is the PR for that:
https://github.com/godotengine/godot/pull/44234

hope that helps!